File: 2606.cpp
#include <iostream>
using namespace std;

int arr[101][101] = {0, };
bool visit[101] = { false, };
int N, M;
int cnt = 0;

void dfs(int s){
	visit[s] = true;

	for(int i = 1; i <= N; i++){
		if(arr[s][i] == 1 && visit[i] == false){
			cnt++;
			//cout << "!" << s << " , " << i << "\n";
			dfs(i);
		}
	}
}

int main(){

	int src, dst;
	cin >> N >> M;

	for(int i = 0; i < M; i++){
		cin >> src >> dst;
		arr[src][dst] = 1;
		arr[dst][src] = 1;
	}

	dfs(1);
	cout << cnt;

	return 0;
}

File: 2606.py
arr = [[0 for i in range(101)] for j in range(101)]
visit = [False for i in range(101)]
com_num = 0
answer = 0

def dfs(num):
    global arr
    global visit
    global com_num
    global answer
    visit[num] = True
    for i in range(1, com_num+1):
        if arr[num][i] == 1 and visit[i] == False:
            answer += 1
            dfs(i)


def main():
    global arr
    global visit
    global com_num
    global answer
    com_num = int(input())
    connect = int(input())
    for i in range(connect):
        src, dest = map(int, input().split())
        arr[src][dest] = 1
        arr[dest][src] = 1
    dfs(1)
    print(answer)
if __name__ == "__main__":
    main()
